### PR TITLE
TCM-430 | Improve `Register` and `Update` flows

### PR DESCRIPTION
**Pull request by VivekV95**
## Description
Improve `Register` and decouple `Update` from authentication

### Summary

- [x] Add First Name and Last Name fields to Register
- [x] Modify design to match specs
- [x] Display Onboarding from `Register`
- [x] Decouple `Update` from auth dlow

### Issue

[TCM-430](https://hyfnla.atlassian.net/browse/TCM-430)

### Video/Screenshot

https://user-images.githubusercontent.com/39810816/120063772-81953500-c02e-11eb-9f99-b032be64b1ea.mov

[Updated header image](https://drive.google.com/file/d/1x-87VaM8p3rYka2DaGfDo4NYlfEQ6KaT/view?usp=sharing)

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/managers/CacheManager.kt [hunk 1]
```diff
@@ -85,21 +84,6 @@ class CacheManager(context: Context) {
         return false
     }
 
-    suspend fun didShowOnboarding(): Boolean {
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/managers/CacheManager.kt:88]: why was this removed?

File: app/src/main/java/com/trx/consumer/screens/login/LoginViewModel.kt [hunk 2]
```diff
@@ -61,11 +61,7 @@ class LoginViewModel @ViewModelInject constructor(
             val response = backendManager.login(email, password)
             eventShowHud.postValue(false)
             if (response.isSuccess) {
-                if (cacheManager.didShowOnboarding()) {
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/screens/login/LoginViewModel.kt:64]: please revert these changes

File: app/src/main/java/com/trx/consumer/screens/register/RegisterFragment.kt [hunk 3]
```diff
@@ -49,7 +50,7 @@ class RegisterFragment : BaseFragment(R.layout.fragment_register) {
 
         viewModel.apply {
             eventLoadView.observe(viewLifecycleOwner, handleLoadView)
-            eventLoadProfile.observe(viewLifecycleOwner, handleLoadProfile)
+            eventLoadOnboarding.observe(viewLifecycleOwner, handleLoadOnboarding)
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/screens/register/RegisterFragment.kt:53]: we use the show keyword for directions load is for screen data loading etc

**Review by onodude — CHANGES_REQUESTED** [review]
looks great overall thank you @VivekV95 added some comments

File: app/src/main/java/com/trx/consumer/managers/CacheManager.kt (hunk 1 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/managers/CacheManager.kt:88]: Will revert

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/managers/CacheManager.kt]: Done, file diff removed so nothing to link

File: app/src/main/java/com/trx/consumer/screens/login/LoginViewModel.kt (hunk 2 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/login/LoginViewModel.kt]: Done, file diff removed so nothing to link

File: app/src/main/java/com/trx/consumer/screens/register/RegisterFragment.kt (hunk 3 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/register/RegisterFragment.kt]: [Done](https://github.com/hyfn/trx-consumer-android/pull/85/files#diff-c4fee2d88a866fbdcba657731ae6cb0e4e05e5f935bd2776cb66b1b9ce023bb7R54)

**Review by onodude — APPROVED** [review]
thank you!
